core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PreAggregateCaseAggregations.java

```diff
@@ -347,9 +349,11 @@ private Optional<CaseAggregation> extractCaseAggregation(Symbol aggregationSymbo
         Symbol projectionSymbol = Symbol.from(aggregation.getArguments().get(0));
         Expression projection = projectNode.getAssignments().get(projectionSymbol);
         Expression unwrappedProjection;
+        boolean wrappedByCast = false;
```

Comment:
@martint do you recall if we have a visitor checking whether expression can throw an exception?

Maybe only then we could wrap expression in `IF`